### PR TITLE
feat(guild): add premium_progress_bar_enabled

### PR DESCRIPTION
**Pull request by ooliver1**
## Summary

discord/discord-api-docs#4001

## Checklist

- [x] I have tested my changes.
- [x] I have updated the documentation to reflect the changes.
- [x] I have run `task pyright` and fixed the relevant issues.

### CODE REVIEW

File: nextcord/guild.py [hunk 1]
```diff
@@ -1697,8 +1712,8 @@ async def edit(
         invites_disabled: :class:`bool`
             Whether the invites should be paused for the guild.
             This will prevent new users from joining said guild.
-
-            .. versionadded:: 2.4
```
Comment (spifory) [inline nextcord/guild.py:1701]: Does this have to be removed?

File: nextcord/guild.py [hunk 2]
```diff
@@ -515,6 +516,8 @@ def _from_data(self, guild: GuildPayload) -> None:
         for event in guild.get("guild_scheduled_events") or []:
             self._store_scheduled_event(event)
 
+        self._premium_progress_bar_enabled: bool = guild.get("premium_progress_bar_enabled", False)
```
Comment (spifory) [inline nextcord/guild.py:519]:
```suggestion
        self._premium_progress_bar_enabled: bool = guild["premium_progress_bar_enabled"]
```
Seeing as this return `true` or `false` and not `null`, the `.get` isn't really needed

File: nextcord/guild.py [hunk 3]
```diff
@@ -1697,8 +1712,8 @@ async def edit(
         invites_disabled: :class:`bool`
             Whether the invites should be paused for the guild.
             This will prevent new users from joining said guild.
-
-            .. versionadded:: 2.4
+        premium_progress_bar_enabled: :class:`bool`
+            Whether the premium guild boost progress bar is enabled in this guild.
```
Comment (spifory) [inline nextcord/guild.py:1716]:
```suggestion
            Whether the premium guild boost progress bar is enabled.
```

**Review by spifory — CHANGES_REQUESTED** [review]

File: nextcord/guild.py (hunk 3 above)
Comment (spifory) [reply nextcord/guild.py:1716]: Alternatively, ```suggestion
Whether the boost progress bar is enabled in this guild.
```

**Review by spifory — APPROVED** [review]

**Review by EmreTech — APPROVED** [review]

**Review by Skelmis — APPROVED** [review]

**Pull request merged**